src/client/Microsoft.Identity.Client/AppConfig/ApplicationConfiguration.cs

```diff
@@ -103,7 +103,7 @@ public string ClientVersion
         internal int ConfidentialClientCredentialCount;
 
         public bool LegacyCacheCompatibilityEnabled { get; internal set; } = true;
-        public bool CacheSynchronizationEnabled { get; internal set; } = true;
+        public bool? CacheSynchronizationEnabled { get; internal set; } = null;

```

Comment:
Hmm, wonder if we can simplify the logic? 
- Keep this line the same - normal `bool` and default to `true`. Since PCA defaults to `true` and there's no way to change the setting neither via PCA options nor via `With` method.
- CCAOptions defaults `EnableCacheSynchronization` to `false` and CCA constructor sets `CacheSynchronizationEnabled` to `false`.
- User can call `WithCacheSynchronization(true)` to change this back, if they want.
- And then we can keep the `TokenCache` as is.

Reply:
I was under the impression that the constructor for CCA would be called after the user calls `WithCacheSynchronization(true)` which would overwrite it, setting it back to false. ill double check

Reply:
Yeah, The CCA constructor is called from BuildConcrete() which happens after the options are configured. i could add it to the `ConfidentialClientApplicationBuilder Create()` method however

Reply:
Hmm, yea, would it make sense to chain`.WithCacheSynchronization(false);` in CCA.Create?

Reply:
Yeah, seems like a good way to do it